storage/include/storage/storage_metrics.h

```diff
@@ -64,24 +58,22 @@ class InMemoryStorageMetrics {
  */
 class RocksDbStorageMetrics {
   concordMetrics::Component rocksdb_comp_;
-  std::unordered_map<::rocksdb::Tickers, concordMetrics::GaugeHandle> active_tickers_;
-  concordMetrics::GaugeHandle total_db_disk_size_;
+  std::unordered_map<::rocksdb::Tickers, concordMetrics::AtomicGaugeHandle> active_tickers_;
+  concordMetrics::AtomicGaugeHandle total_db_disk_size_;
 
   std::shared_ptr<::rocksdb::SstFileManager> sstFm;
   std::shared_ptr<::rocksdb::Statistics> statistics;
-
-  std::chrono::seconds metrics_update_interval_ = std::chrono::seconds(5);  // TODO: move to configuration
-  std::chrono::steady_clock::time_point last_metrics_update_ = std::chrono::steady_clock::now();
+  std::unique_ptr<concord::util::PeriodicCall> update_metrics_ = nullptr;
```

Comment:
I was thinking that when we run the `InMemoryStorageMetrics`  and `RocksDbStorageMetrics` destructors, we want to ensure that the thread stops before we destruct metrics members as the thread might still call them. Therefore, I think it might make sense to put an explicit comment saying that the `PeriodicCall` pointer should be the last member in these classes, ensuring the thread has stopped. What do you think? Did I get it correctly?

Reply:
I have added a comment